### ISSUE TITLE
Refactor processing of handlers

### ISSUE THREAD

**Issue by strekm** [kyma-project/api-gateway#95]
**Description**

Introducing istio based jwt handler in parallel to oathkeeper handler exposed need to refactor source code in `processing.go` file. Break processing of handlers into separate files including unit tests. 

ACs:
- [x] slimmer `processing.go`
- [x] introduced structure should support adding handlers
- [x] It is possible to have different object kinds processed based on the handler
- [x] It is possible to have different processing logic of an object based on the handler
- [x] unit tests are not too sensitive to changes
- [x] component tests and manual tests are green

**Reasons**

Simplifying source code and improving maintenance

**DoD:**
~~- [ ] provide documentation~~
~~- [ ] publish release notes and _What's New_ updates for Kyma customers~~
- [x] provide unit tests
- [x] test on SKR
- [x] verify [Get started guides](https://kyma-project.io/docs/kyma/latest/02-get-started/) 
~~- [ ] verify resource limits~~
- [ ] followup issue
~~- [ ] create release and bump in kyma~~

**Comment by triffer** [comment]
**PR:**
- https://github.com/kyma-project/api-gateway/pull/99
